src/App.tsx

```diff
@@ -1,56 +1,68 @@
-import React from 'react';
+import React, { useMemo, useState, useCallback } from 'react';
 import './App.scss';
+import classNames from 'classnames';
 import { peopleFromServer } from './data/people';
+import { PeopleList } from './components/PeopleList';
+import { Person } from './types/Person';
+import { debounce } from './services/debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [person, setPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+
+  const applyQuery = useCallback(debounce(setQuery, 1000), []);
+
+  const filteredPerson = useMemo(() => {
+    return peopleFromServer
+      .filter(human => human.name.toLowerCase().includes(query.toLowerCase()));
+  }, [query]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
```

Comment:
Do you really need to call this setter here?
If you do it your debounce `applyQuery` wouldn't work.
Check it, please